src/Http/GraphRequest.php

```diff
@@ -321,31 +321,35 @@ function ($reason) {
     */
     public function download($path, $client = null)
     {
+        //Add custom error handler to catch file I/O warnings
+        set_error_handler(function() {});
+
         if (is_null($client)) {
             $client = $this->createGuzzleClient();
         }
         try {
-            if (file_exists($path) && is_writeable($path)) {
-                $file = fopen($path, 'w');
-
-                $client->request(
-                    $this->requestType, 
-                    $this->_getRequestUrl(), 
-                    [
-                        'body' => $this->requestBody,
-                        'sink' => $file
-                    ]
-                );
-                if(is_resource($file)){
-                    fclose($file);
-                }
-            } else {
+            $file = fopen($path, 'w');
+            if (!$file) {
                 throw new GraphException(GraphConstants::INVALID_FILE);
             }
+
+            $client->request(
+                $this->requestType, 
+                $this->_getRequestUrl(), 
+                [
+                    'body' => $this->requestBody,
+                    'sink' => $file
+                ]
+            );
+            if(is_resource($file)){
+                fclose($file);
+            }
             
         } catch(GraphException $e) {
             throw new GraphException(GraphConstants::INVALID_FILE);
         }
+        //Restore default error handler
+        restore_error_handler();
```

Comment:
This should be in a `finally` block to prevent it never getting removed in case of exception.